src/v_fmt.c

```diff
@@ -0,0 +1,481 @@
+        }
+
+        // Add the column data
+        array_push(columns, col);
+
+        // Go to next column
+        ++offset;
+    }
+
+    size_t size = 0;
+
+    // Calculate needed memory size to allocate patch buffer
+    size += 4 * sizeof(int16_t);                   // 4 header shorts
+    size += array_size(columns) * sizeof(int32_t); // offsets table
+
+    for (int c = 0; c < array_size(columns); c++)
+    {
+        for (int p = 0; p < array_size(columns[c].posts); p++)
+        {
+            size_t post_len = 0;
+
+            post_len += 2; // two bytes for post header
+            post_len += 1; // dummy byte
+            post_len += array_size(columns[c].posts[p].pixels); // pixels
+            post_len += 1; // dummy byte
+
+            size += post_len;
+        }
+
+        size += 1; // room for 0xff cap byte
+    }
+
+    byte *output = Z_Malloc(size, tag, user);
+    byte *rover = output;
+
+    // write header fields
+    PUTSHORT(rover, width);
+    PUTSHORT(rover, height);
+    // This is written to afterwards
+    PUTSHORT(rover, 0);
+    PUTSHORT(rover, 0);
+
+    // set starting position of column offsets table, and skip over it
+    byte *col_offsets = rover;
+    rover += array_size(columns) * 4;
+
+    for (int c = 0; c < array_size(columns); c++)
+    {
+        // write column offset to offset table
+        uint32_t offs = (uint32_t)(rover - output);
+        PUTLONG(col_offsets, offs);
+
+        // write column posts
+        for (int p = 0; p < array_size(columns[c].posts); p++)
+        {
+            // Write row offset
+            PUTBYTE(rover, columns[c].posts[p].row_off);
+
+            // Write number of pixels
+            int numpixels = array_size(columns[c].posts[p].pixels);
+            PUTBYTE(rover, numpixels);
+
+            // Write pad byte
+            byte lastval = numpixels ? columns[c].posts[p].pixels[0] : 0;
+            PUTBYTE(rover, lastval);
+
+            // Write pixels
+            for (int a = 0; a < numpixels; a++)
+            {
+                lastval = columns[c].posts[p].pixels[a];
+                PUTBYTE(rover, lastval);
+            }
+
+            // Write pad byte
+            PUTBYTE(rover, lastval);
+
+            array_free(columns[c].posts[p].pixels);
+        }
+
+        // Write 255 cap byte
+        PUTBYTE(rover, 0xff);
+
+        array_free(columns[c].posts);
+    }
+
+    array_free(columns);
+
+    // Done!
+    return (patch_t *)output;
+}
+
+static patch_t *DummyPatch(int lump, pu_tag tag)
+{
+    int num = (W_CheckNumForName)("TNT1A0", ns_sprites);
+    int len = W_LumpLength(num);
+    patch_t *dummy = W_CacheLumpNum(num, PU_CACHE);
+
+    Z_Malloc(len, tag, &lumpcache[lump]);
+    memcpy(lumpcache[lump], dummy, len);
+    return lumpcache[lump];
+}
+
+static void *DummyFlat(int lump, pu_tag tag)
+{
+    const int size = 64 * 64;
+
+    Z_Malloc(size, tag, &lumpcache[lump]);
+    memset(lumpcache[lump], v_darkest_color, size);
```

Comment:
Could still add an 8x8 checkerboard pattern here.

Reply:
I think we can add it later as an internal resource. There is no reason to generate it in code.

Reply:
Ok.